src/inmanta/data/__init__.py

```diff
@@ -5498,14 +5500,20 @@ async def get_versions(cls, environment: uuid.UUID, start: int = 0, limit: int =
         return versions
 
     async def delete_cascade(self) -> None:
-        async with self.get_connection() as con:
-            async with con.transaction():
-                # Delete all code associated with this version
-                await Code.delete_all(connection=con, environment=self.environment, version=self.version)
-
-                # Delete ConfigurationModel and cascade delete on connected tables
-                await self.delete(connection=con)
+        await Compile.delete_all(environment=self.environment, version=self.version)  # Triggers cascading delete report table
+        await Code.delete_all(environment=self.environment, version=self.version)
+        await DryRun.delete_all(environment=self.environment, model=self.version)
+        await UnknownParameter.delete_all(environment=self.environment, version=self.version)
+        await self._execute_query(
+            "DELETE FROM public.resourceaction_resource WHERE environment=$1 AND resource_version=$2",
+            self.environment,
+            self.version,
+        )
+        await ResourceAction.delete_all(environment=self.environment, version=self.version)
+        await Resource.delete_all(environment=self.environment, model=self.version)
+        await self.delete()
 
+        async with self.get_connection() as con:
```

Comment:
Could we have a nullable foreign key in `Parameter` referencing `Resource.resource_id` and rely on a cascading delete of Resource instead of this ?

Reply:
No, we can't. A foreign key always references one specific record in a certain table. The primary key of the Resource table is (environment, model, resource_id). The same resource_id might occur in multiple version of the configurationmodel and as such there will be multiple records in the resource table for this resource_id. So adding a foreign key is not possible.

Reply:
I see, thanks :+1: